subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileToolchainIntegrationTest.groovy

```diff
@@ -256,7 +256,7 @@ public class Foo {
             .runWithFailure()
 
         then:
-        failureHasCause('The new Java toolchain feature cannot be used at the project level in combination with source and/or target compatibility')
+        failureHasCause("Error while evaluating property 'sourceCompatibility' of task ':compileJava': The new Java toolchain feature cannot be used at the project level in combination with source and/or target compatibility")
```

Comment:
I wonder if these would be better as an embedded cause instead.

Reply:
Those things are really user-facing error messages. You don't see the embedded cause if you don't use `--stacktrace` I think. Or does this require some gymnastics with `@Contextual`?